n3fit/src/evolven3fit/eko_utils.py

```diff
@@ -0,0 +1,55 @@
+from validphys.loader import Loader
+from . import utils
+from ekobox import gen_theory, gen_op
+from eko import run_dglap
+
+
+def construct_eko_cards(theoryID, op_card_dict, t_card_dict, q_fin, q_points, x_grid):
+    """Return the theory and operator cards used to construct the eko"""
+    # theory_card construction
+    theory = Loader().check_theoryID(theoryID).get_description()
+    theory.pop("FNS")
+    theory.update(t_card_dict)
+    t_card = gen_theory.gen_theory_card(theory["PTO"], theory["Q0"], update=theory)
+    # construct operator card
+    q2_grid = utils.generate_q2grid(
+        theory["Q0"],
+        q_fin,
+        q_points,
+        {theory["mb"]: theory["kbThr"], theory["mt"]: theory["ktThr"]},
+    )
+    op_card = gen_op.gen_op_card(q2_grid, update={"interpolation_xgrid": x_grid})
+    op_card.update(op_card_dict)
+    return t_card, op_card
```

Comment:
Are these two cards accessible inside the `eko`?

If not, maybe they should be output as well (and then stored together with the theory for instance). If they are already accessible within the `eko.tar` produced I would highly appreciate a snippet in the `docstr` on how to access it:

```python
Example
------------
> from eko import read_eko
> read_eko("/path/toeko.tar").theory_card
```
or whatever.

Reply:
Actually, inside the eko there are some of the info contained in the theory and in the operator cards but not all of them. However, I don't think the info that are not in the eko to be very relevant but it is true that you need them in order to reproduce the very same eko (correct me if I am wrong @AleCandido @felixhekhorn). So maybe it is worth to store them in the theory.

Reply:
We're going to change, and indeed only the values consumed will be retained (actually, I'd like to refuse a card containing further information, but I'm not sure about this...).
But at the moment, the full cards should be stored inside the operator itself.

Reply:
But at the moment the opcard and theorycard are not contained in the eko right?

Reply:
Depends what you mean by eko. If it's `eko.tar` I believe they are.

In any case, it is called `<whatever>.tar`, and not `.eko` (even though I'm still considering the `.eko.tar` option), exactly because we wanted to stress that you can have a look inside in an intuitive way.

Reply:
But maybe you're right, it's not yet on `master`:
https://github.com/N3PDF/eko/blob/01efee1d27d215e895dd055debff8905af3d548c/src/eko/output.py#L299-L300

Reply:
Ok so this is the reason why I did not find them :)